Basic/DistributedAuthoritySocialHub/Assets/Scripts/Services/Vivox3DPositioning.cs

```diff
@@ -0,0 +1,49 @@
+using UnityEngine;
+using Unity.Netcode;
+using Unity.Multiplayer.Samples.SocialHub.GameManagement;
+
+namespace Unity.Multiplayer.Samples.SocialHub.Services
+{
+    class Vivox3DPositioning : NetworkBehaviour
+    {
+        bool m_Initialized;
+        float m_NextPosUpdate;
+
+        void Start()
+        {
+            GameplayEventHandler.OnChatIsReady -= OnChatIsReady;
+            GameplayEventHandler.OnChatIsReady += OnChatIsReady;
+
+            GameplayEventHandler.OnExitedSession -= OnExitSession;
+            GameplayEventHandler.OnExitedSession += OnExitSession;
+        }
```

Comment:
This needs to be done only on the instance that has authority, correct?
So, I would suggest doing the subscriptions on OnNetworkSpawn & similarly clearing on OnNetworkDespawn.
We can also hook into Netcode's NetworkUpdateStage too, and only have the authoritative instance subscribe to that callback. Check out AvatarTransform, you can subscribe to most update loops.

Reply:
Good idea thanks, i changed it

Reply:
I would still look into adding the NetworkUpdateStage hook too, since only the authoritative instance has the need for an update loop.

Reply:
The behavior is switched of altogether if it has no authority. So update should not run.

```
 public override void OnNetworkSpawn()
        {
            base.OnNetworkSpawn();
            if (!HasAuthority)
            {
                enabled = false; // Disable the Component if it is on the current player Avatar
                return;
            }
```

Reply:
That works too!